core/src/main/scala/org/apache/spark/scheduler/MapStatus.scala

```diff
@@ -166,16 +165,26 @@ private[spark] class HighlyCompressedMapStatus private (
     loc.writeExternal(out)
     emptyBlocks.writeExternal(out)
     out.writeLong(avgSize)
-    out.writeObject(hugeBlockSizesArray)
+    out.writeInt(hugeBlockSizes.size)
+    hugeBlockSizes.foreach { kv =>
+      out.writeInt(kv._1)
+      out.writeByte(kv._2)
+    }
   }
 
   override def readExternal(in: ObjectInput): Unit = Utils.tryOrIOException {
     loc = BlockManagerId(in)
     emptyBlocks = new RoaringBitmap()
     emptyBlocks.readExternal(in)
     avgSize = in.readLong()
-    hugeBlockSizesArray = in.readObject().asInstanceOf[Array[Tuple2[Int, Byte]]]
-    hugeBlockSizes = hugeBlockSizesArray.toMap
+    val count = in.readInt()
+    val tmpHugeBlockSizes = mutable.ArrayBuffer[Tuple2[Int, Byte]]()
+    (0 until count).foreach { _ =>
+      val block = in.readInt()
+      val size = in.readByte()
+      tmpHugeBlockSizes += Tuple2(block, size)
+    }
+    hugeBlockSizes = tmpHugeBlockSizes.toMap
```

Comment:
Is it better to use ArrayBuffer, and then ArrayBuffer -> toMap compared to mutable.HashMap and then .toMap ?